### PR TITLE
Create view model factory

### PR DESCRIPTION
**Pull request by hodamohammadi**
creating a view model factory singleton to create instances of view models

### CODE REVIEW

File: conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt [hunk 1]
```diff
@@ -21,7 +24,8 @@ class SingleChatFragment : Fragment(), MessageInput.InputListener, MessageInput.
         MessageInput.TypingListener {
 
     private lateinit var messagesAdapter: MessagesListAdapter<IMessage>
-
+    private lateinit var chatViewModel: ChatViewModel
+
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/1190adb857090d59c52d8c8413a61bee78f91f96/conveyor/src/main/java/com/hodamohammadi/conveyor/fragments/SingleChatFragment.kt#L28"></span>Trailing space(s)</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+
+/**
+ * View model for communication between chat UIs.
+ */
+class ChatViewModel: ViewModel() {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/c4865c7bb4f2ca24349c9a3cfab0a20e4605c164/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ChatViewModel.kt#L8"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

File: conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.hodamohammadi.conveyor.viewmodels
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+
+/**
+ * singleton of view model factory to create instances of view models.
+ */
+object ViewModelFactory: ViewModelProvider.Factory {
```
Comment (ReviewerOfCode) [inline conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/hodamohammadi/Conveyor/blob/c4865c7bb4f2ca24349c9a3cfab0a20e4605c164/conveyor/src/main/java/com/hodamohammadi/conveyor/viewmodels/ViewModelFactory.kt#L9"></span>Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

**Comment by ReviewerOfCode** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/c4865c7bb4f2ca24349c9a3cfab0a20e4605c164/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L16">conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultMessage.kt#L16</a> - Missing spacing before “:”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/c4865c7bb4f2ca24349c9a3cfab0a20e4605c164/conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L13">conveyor/src/main/java/com/hodamohammadi/conveyor/models/DefaultUser.kt#L13</a> - Missing spacing before “:”</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/hodamohammadi/Conveyor/blob/c4865c7bb4f2ca24349c9a3cfab0a20e4605c164/conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5">conveyor/src/main/java/com/hodamohammadi/conveyor/services/ConveyorNotificationService.kt#L5</a> - Missing spacing before “:”</td>
    </tr>
  </tbody>
</table>

### AndroidLint found issues

#### Warning (2)

| File | Line | Reason |
| ---- | ---- | ------ |
`conveyor/build.gradle` | 12 | Not targeting the latest versions of Android; compatibility modes apply. Consider testing and updating this version. Consult the android.os.Build.VERSION_CODES javadoc for details. 
`conveyor/src/main/AndroidManifest.xml` | 6 | On SDK version 23 and up, your app data will be automatically backed up, and restored on app install. Your GCM regid will not work across restores, so you must ensure that it is excluded from the back-up set. Use the attribute `android:fullBackupContent` to specify an `@xml` resource which configures which files to backup. More info: https://developer.android.com/training/backup/autosyncapi.html 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by itsAlexNguyen — APPROVED** [review]

**Pull request merged**